arch/arm/src/sam34/sam_dmac.c

```diff
@@ -1380,9 +1341,9 @@ void weak_function arm_dma_initialize(void)
 
   putreg32(DMAC_EN_ENABLE, SAM_DMAC_EN);
 
-  /* Initialize semaphores */
+  /* Initialize mutex & semaphores */
 
-  nxsem_init(&g_chsem, 0, 1);
+  nxmutex_init(&g_chlock);
   nxsem_init(&g_dsem, 0, CONFIG_SAM34_NLLDESC);
```

Comment:
```suggestion
  nxsem_init(&g_dsem, 0, CONFIG_SAM34_NLLDESC);
  nxsem_set_protocol(&g_dsem, SEM_PRIO_NONE);
```
Optionally: we can do static init for both `g_dsem` and `g_chlock`

Reply:
Done.

Reply:
Fixed in static initializer.